src/bit-systems/link-hover-menu.ts

```diff
@@ -0,0 +1,51 @@
+import { defineQuery, enterQuery, entityExists } from "bitecs";
+import type { HubsWorld } from "../app";
+import { Interacted, Link, LinkHoverMenu, LinkHoverMenuItem, HoveredRemoteRight } from "../bit-components";
+import { anyEntityWith } from "../utils/bit-utils";
+
+const NULL_EID = 0;
+
+const hoveredQuery = defineQuery([HoveredRemoteRight, Link]);
+const hoveredEnterQuery = enterQuery(hoveredQuery);
+const hoveredMenuItemQuery = defineQuery([HoveredRemoteRight, LinkHoverMenuItem]);
+const clickedMenuItemQuery = defineQuery([Interacted, LinkHoverMenuItem]);
+
+export function linkHoverMenuSystem(world: HubsWorld) {
+  // Assumes always only single LinkHoverMenu entity exists.
+  const menuEid = anyEntityWith(world, LinkHoverMenu)!;
+  const menuObject = world.eid2obj.get(menuEid)!;
+
+  // Save Link object eid in LinkHoverMenu when hovered.
+  hoveredEnterQuery(world).forEach((eid: number) => {
+    LinkHoverMenu.targetObject[menuEid] = eid;
+    const targetObject = world.eid2obj.get(eid)!;
+    targetObject.add(menuObject);
+  });
+
+  // Check if the cursor it hovered on Link object or Link menu button object.
+  const hovered = hoveredQuery(world).length > 0 || hoveredMenuItemQuery(world).length > 0;
+
+  if (hovered) {
+    // Hovered then make the menu visible and handle clicks if needed.
+    menuObject.visible = true;
+    if (clickedMenuItemQuery(world).length > 0) {
+      const linkEid = LinkHoverMenu.targetObject[menuEid];
+      if (linkEid !== NULL_EID) {
+        // Just in case
```

Comment:
I don't like that we got as far as being able to click on something for an object that was deleted. Ideally we can clean up the control flow so that when the target of this menu is deleted we wouldn't even get here.

Reply:
Yes, true. I hope we can work on more proper interactive flow in next cycle.